_studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp

```diff
@@ -1384,6 +1384,33 @@ void VideoDECODEH265::FillOutputSurface(mfxFrameSurface1 **surf_out, mfxFrameSur
             info->FrameType |= MFX_FRAMETYPE_REF;
    }
 
+    mfxExtMasteringDisplayColourVolume* dispaly_colour = (mfxExtMasteringDisplayColourVolume*)GetExtendedBuffer(surface_out->Data.ExtParam, surface_out->Data.NumExtParam, MFX_EXTBUFF_MASTERING_DISPLAY_COLOUR_VOLUME);
+    if (pFrame->m_mastering_display.payLoadSize != 0)
+    {
+        if (dispaly_colour && pFrame->m_mastering_display.payLoadSize > 0)
+        {
+            dispaly_colour->InsertPayloadToggle = MFX_PAYLOAD_IDR;
```

Comment:
Could you explain why it's always MFX_PAYLOAD_IDR ?

Reply:
If the frames do not have `MasteringDisplayColourVolume` and `ContentLightLevelInfo` SEI, **InsertPayloadToggle** is default set as `0` in output surface during decoding steps, this SEI will not be used in encoding steps.

And if parse these two SEIs during decoding steps, we need to set the  **InsertPayloadToggle** to `MFX_PAYLOAD_IDR`, then this SEI can be used in encoding steps.

Refers to [mfxstructures.h](https://github.com/intel-innersource/drivers.gpu.media.sdk.lib/blob/86b711813aef29045e36c374792d84aaf37996cc/api/vpl/mfxstructures.h#L2804)
>  If the application attaches this structure to the mfxEncodeCtrl structure at runtime, the encoder inserts the HDR SEI message for the current frame and ignores InsertPayloadToggle.
If the application attaches this structure to the mfxVideoParam structure during initialization or reset, the encoder inserts the HDR SEI message based on InsertPayloadToggle.

Reply:
But it's an encoder input flag. It doesn't seems that decoder should control encoder behavior.

Reply:
Do you mean the encoder will set this flag to decide which frame needs to use HDR SEI?

Reply:
It's discussible. I think a user should do it. But there might be other options like the one you proposed. Anyway, we need to get aligned on how users will operate with the feature.

Reply:
I wonder whether this is even worth any discussion:

- For decoding, the two new structures are provided as part of the frame data
- For encoding, the data needs to be either attached once at init to mfxVideoParam (constant values which are repeated) or to mfxEncodeCtrl for each frame that is submitted for encoding
- As the SDK consumer doesn't control the lifetime of the memory for those structures coming from the decoder, it cannot simply use these structures as-is and attach them to the encoding parameters.
  That means: it will be required to allocate new structures and copy the individual values manually - so it's all up to the developer's coding and neither the flag nor even the structures will automatically be propagated to the encoder

This looks pretty good to me already..

Reply:
Thanks for the advice, we will soon close the discussion and then update this PR. 
@softworkz, can you provide some HEVC clips contain both mastering_display and content_Light_Level SEI? So i can do some tests for this RP, thank you in advance~

Reply:
You can try one of those:
- https://4kmedia.org/ultra-hd-hdr-samsung-4k-demo-wonderland/
- https://4kmedia.org/sony-swordsmith-hdr-uhd-4k-demo/

They have Mastering Display data for sure, CLL most likely but don't remember exactly.